### PR TITLE
build: cloud-first ksqldb releases

### PR DESCRIPTION
**Pull request by vcrfxia**
### Description 

This PR updates the release script to include building ksqlDB images for Confluent Cloud, as described in https://confluentinc.atlassian.net/wiki/spaces/KSQL/pages/1047173795/Cloud-first+ksqlDB+releases.

Highlights include:
- Release script now builds both on-prem and cloud images for all (non-PR job) runs
- Updated image tags
- Non-release jobs now create GitHub tags and push maven artifacts as well as release jobs

The new parts of the script for building the cloud image are largely copy-pasted from the existing parts for the on-prem image. The cloud parts will be moved into cc-docker-ksql in a follow-up PR.

### Testing done 

Manual verification of GitHub tagging and image building.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -319,13 +334,162 @@ def job = {
...
+    }
+
+    stage('Build CCloud KSQL Docker image') {
+        dir('ksqldb-ccloud-docker') {
+            archiveArtifacts artifacts: 'pom.xml'
+            withCredentials([
+                usernamePassword(credentialsId: 'JenkinsArtifactoryAccessToken', passwordVariable: 'ARTIFACTORY_PASSWORD', usernameVariable: 'ARTIFACTORY_USERNAME'),
+                usernameColonPassword(credentialsId: 'Jenkins GitHub Account', variable: 'GIT_CREDENTIAL')]) {
+                    withDockerServer([uri: dockerHost()]) {
+                        withMaven(globalMavenSettingsFilePath: settingsFile, options: mavenOptions) {
```
Comment (elismaga) [inline Jenkinsfile]: @vcrfxia this is your issue right here. this is still configured to point at the beta maven repo. up above where it creates the settingsFile, it configures it to the beta repo. so if you are just hacking something together, create the file again but change it to the regular repo where all the snapshots are

Comment (vcrfxia) [reply Jenkinsfile]: Discussed offline: turns out it's not possible to use both the default Jenkins credentials configured in Jenkins itself and a custom maven settings file, so the workaround we found was to configure the additional repos (ksqldb staging repo and beta maven repo, needed for dependencies) in the cc-docker-ksql pom itself, and use the default Jenkins credentials here in the Jenkinsfile.

**Comment by vcrfxia** [comment]
This pipeline is now working and ready for review.

As @elismaga and I discussed offline, it'd be nice split this job into two jobs (one for building the on-prem image and one for building the cloud image) and have one trigger the other, but I'd like to do that in a follow-up PR as we'd like to start using an early version of this pipeline soon (within the next few days).

**Comment by vcrfxia** [comment]
Closing in favor of https://github.com/confluentinc/ksql/pull/4876 which splits this job into two as discussed above.

**Pull request closed without merging**